chia/full_node/mempool_manager.py

```diff
@@ -506,27 +507,28 @@ async def new_peak(self, new_peak: Optional[BlockRecord]) -> List[Tuple[SpendBun
         self.peak = new_peak
 
         old_pool = self.mempool
-        self.mempool = Mempool(self.mempool_max_total_cost)
+        async with self.lock:
```

Comment:
There are no changes here apart from indenting and adding this lock

Reply:
as long as all updates to the mempool are made in here or in `respond_transaction()` that seems safe. However, when we farm a block ourselves, don't we remove transactions out of the mempool? Or do we let the block go through the normal path and then come back here if it's accepted by the chain?

Reply:
Normal path, exactly.